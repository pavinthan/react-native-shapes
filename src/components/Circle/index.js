import Circle from './Circle';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Circle.propTypes = propTypes;
Circle.defaultProps = defaultProps;

export default Circle;
