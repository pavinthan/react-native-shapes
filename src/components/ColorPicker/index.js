import ColorPicker from './ColorPicker';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

ColorPicker.propTypes = propTypes;
ColorPicker.defaultProps = defaultProps;

export default ColorPicker;
