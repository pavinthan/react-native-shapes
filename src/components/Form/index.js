import Form from './Form';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Form.propTypes = propTypes;
Form.defaultProps = defaultProps;

export default Form;
