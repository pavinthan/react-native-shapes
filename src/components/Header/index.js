import Header from './Header';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

export default Header;
