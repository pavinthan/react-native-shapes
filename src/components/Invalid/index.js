import Invalid from './Invalid';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Invalid.propTypes = propTypes;
Invalid.defaultProps = defaultProps;

export default Invalid;
