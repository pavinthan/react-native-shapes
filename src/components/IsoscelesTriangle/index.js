import IsoscelesTriangle from './IsoscelesTriangle';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

IsoscelesTriangle.propTypes = propTypes;
IsoscelesTriangle.defaultProps = defaultProps;

export default IsoscelesTriangle;
