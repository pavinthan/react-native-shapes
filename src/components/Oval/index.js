import Oval from './Oval';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Oval.propTypes = propTypes;
Oval.defaultProps = defaultProps;

export default Oval;
