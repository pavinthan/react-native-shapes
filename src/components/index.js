export { default as Circle } from './Circle';
export { default as ColorPicker } from './ColorPicker';
export { default as Form } from './Form';
export { default as Header } from './Header';
export { default as Invalid } from './Invalid';
export { default as IsoscelesTriangle } from './IsoscelesTriangle';
export { default as Oval } from './Oval';
export { default as Parallelogram } from './Parallelogram';
export { default as Polygon } from './Polygon';
export { default as Rectangle } from './Rectangle';
export { default as Whiteboard } from './Whiteboard';
