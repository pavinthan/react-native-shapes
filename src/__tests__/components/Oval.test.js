import React from 'react';
import { Oval } from '../../components';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const rendered = renderer
    .create(<Oval width={100} height={200} fill="#000" />)
    .toJSON();
  expect(rendered).toBeTruthy();
});

it('renders square component', () => {
  const tree = renderer
    .create(<Oval width={100} height={200} fill="#000" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
