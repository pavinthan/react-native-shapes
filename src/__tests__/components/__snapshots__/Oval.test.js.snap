// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders square component 1`] = `
<RNSVGSvgView
  bbHeight="400"
  bbWidth="200"
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
      false,
      Object {
        "flex": 0,
        "height": 400,
        "width": 200,
      },
    ]
  }
>
  <RNSVGEllipse
    cx="100"
    cy="200"
    fill={
      Array [
        0,
        0,
        0,
        0,
        1,
      ]
    }
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    onLayout={undefined}
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "fill",
      ]
    }
    rotation={0}
    rx="50"
    ry="100"
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={null}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="1"
    x={0}
    y={0}
  />
</RNSVGSvgView>
`;
